client/state/data-layer/wpcom/sites/comments/index.js

```diff
@@ -54,13 +53,16 @@ const changeCommentStatus = ( { dispatch, getState }, action ) => {
 			}
 		)
 	);
+
+	dispatch( { type: COMMENTS_COUNT_INCREMENT, siteId, postId, status } );
+	dispatch( { type: COMMENTS_COUNT_DECREMENT, siteId, postId, status: previousStatus } );
```

Comment:
is there no good way for the relevant reducers to listen for `COMMENTS_CHANGE_STATUS` actions and update their counts accordingly?

Reply:
It's not as clean: see https://github.com/Automattic/wp-calypso/pull/20809

Reply:
The tricky part is I'd either need to look up `previousStatus` in the reducer, or update the action shape to include it.

Reply:
personally I think it's perfectly fine to add code into the reducer and even look at `previousStatus`. it's either govern the data type in its own reducer or add more reducers `¯\_(ツ)_/¯`

/me don't care too much because obviously our norm is just add more reducers…